NitroxServer/Communication/Packets/Processors/WeldActionProcessor.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NitroxModel.Logger;
+using NitroxModel.Packets;
+using NitroxServer.Communication.Packets.Processors.Abstract;
+using NitroxServer.GameLogic;
+
+namespace NitroxServer.Communication.Packets.Processors
+{
+    class WeldActionProcessor : AuthenticatedPacketProcessor<WeldAction>
+    {
+        private readonly SimulationOwnershipData simulationOwnershipData;
+
+        public WeldActionProcessor(SimulationOwnershipData simulationOwnershipData)
+        {
+            this.simulationOwnershipData = simulationOwnershipData;
+        }
+
+        public override void Process(WeldAction packet, Player player)
+        {
+            Player simulatingPlayer = simulationOwnershipData.GetPlayerForLock(packet.Id);
+
+            if (simulatingPlayer != null)
+            {
+                Log.Debug($"Send WeldAction to simulating player {simulatingPlayer.Name} for entity {packet.Id}");
+                simulatingPlayer.SendPacket(packet);
+            }
```

Comment:
else, should we assign simulation the requester and send them back the packet?

Reply:
Due to the initial simulation ownership and changed simulation ownership when a player disconnects, it should not be possible that no player has the simulation of an vehicle. Therefore, no assign in retrospect is needed.

Reply:
Let's replace with a Validate.NotNull() instead of null check then.